subsys/net/lib/nrf_cloud/src/nrf_cloud_fota_support.c

```diff
@@ -0,0 +1,554 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/settings/settings.h>
+#include <net/nrf_cloud.h>
+#if defined(CONFIG_NRF_CLOUD_COAP)
+#include <net/nrf_cloud_coap.h>
+#endif
+#include <net/nrf_cloud_rest.h>
+#include <zephyr/logging/log.h>
+#include <net/fota_download.h>
+
+LOG_MODULE_REGISTER(nrf_cloud_fota_support, CONFIG_NRF_CLOUD_FOTA_SUPPORT_LOG_LEVEL);
+
+/* Use the settings library to store FOTA job information to flash so
+ * that the job status can be updated after a reboot
+ */
+#if defined(CONFIG_FOTA_USE_NRF_CLOUD_SETTINGS_AREA)
+#define FOTA_SETTINGS_NAME		NRF_CLOUD_SETTINGS_FULL_FOTA
+#define FOTA_SETTINGS_KEY_PENDING_JOB	NRF_CLOUD_SETTINGS_FOTA_JOB
+#define FOTA_SETTINGS_FULL		NRF_CLOUD_SETTINGS_FULL_FOTA_JOB
+#else
+#define FOTA_SETTINGS_NAME		CONFIG_FOTA_SETTINGS_NAME
+#define FOTA_SETTINGS_KEY_PENDING_JOB	CONFIG_FOTA_SETTINGS_KEY_PENDING_JOB
+#define FOTA_SETTINGS_FULL		FOTA_SETTINGS_NAME "/" FOTA_SETTINGS_KEY_PENDING_JOB
+#endif
```

Comment:
Should this be customizable specifically for these helpers?

I argue this is one of two built-in implementations of the FOTA job storage system, they should just straight up use the same KConfigs